File: with_elixir/test/day2_test.exs
defmodule Day2Test do
  use ExUnit.Case
  doctest Day2

  import ReadInput

  test "part1..." do
    assert Day2.part1(read_input("day2sample")) == 2
  end

  test "part2..." do
    assert Day2.part2(read_input("day2sample")) == 4
  end
end

File: with_node/src/day2.test.ts
import assert from "node:assert";
import test from "node:test";

import { readInput } from "./readInput.ts";
import { part1, part2 } from "./day2.ts";

test("day2...", async (t) => {
  await t.test("part1 says...", () => {
    assert.strictEqual(part1(readInput("day2sample")), 2);
  });

  await t.test("part2 says...", () => {
    assert.strictEqual(part2(readInput("day2sample")), 4);
  });
});
